runtime/drivers/duckdb/transporter/objectStore_to_duckDB.go

```diff
@@ -260,3 +267,53 @@ func (a *appender) scanSchemaFromQuery(ctx context.Context, qry string) (map[str
 
 	return schema, nil
 }
+
+func (t *objectStoreToDuckDB) ingestDuckDBSQL(
+	ctx context.Context,
+	originalSQL string,
+	iterator drivers.FileIterator,
+	dbSink *drivers.DatabaseSink,
+	opts *drivers.TransferOpts,
+	p drivers.Progress,
+) error {
+	allFiles := make([]string, 0)
+	for iterator.HasNext() {
+		files, err := iterator.NextBatch(opts.IteratorBatch)
+		if err != nil {
+			return err
+		}
+		allFiles = append(allFiles, files...)
+	}
+
+	ast, err := duckdbsql.Parse(originalSQL)
+	if err != nil {
+		return err
+	}
+
+	err = ast.RewriteTableRefs(func(table *duckdbsql.TableRef) (*duckdbsql.TableRef, bool) {
```

Comment:
Maybe put a comment that upstream logic is expected to guarantee that there's only one table ref in the SQL

Reply:
Or safer – add a check here that returns an error if more than one table reference was matched